cmd/kubeadm/app/util/etcd/etcd_test.go

```diff
@@ -351,3 +352,141 @@ func TestGetRawEtcdEndpointsFromPodAnnotationWithoutRetry(t *testing.T) {
 		})
 	}
 }
+
+type fakeEtcdClient struct {
+	members   []*pb.Member
+	endpoints []string
+}
+
+// Close shuts down the client's etcd connections.
+func (f *fakeEtcdClient) Close() error {
+	f.members = []*pb.Member{}
+	return nil
+}
+
+// Endpoints lists the registered endpoints for the client.
+func (f *fakeEtcdClient) Endpoints() []string {
+	return f.endpoints
+}
+
+// MemberList lists the current cluster membership.
+func (f *fakeEtcdClient) MemberList(_ context.Context) (*clientv3.MemberListResponse, error) {
+	return &clientv3.MemberListResponse{
+		Members: f.members,
+	}, nil
+}
+
+// MemberAdd adds a new member into the cluster.
+func (f *fakeEtcdClient) MemberAdd(_ context.Context, peerAddrs []string) (*clientv3.MemberAddResponse, error) {
+	return nil, fmt.Errorf("not implemented")
```

Comment:
these should be `github.com/pkg/errors.New(..)`
for consistency.

EDIT: we can define a file-local not implemented error var and return it where needed.

Reply:
Fixed by https://github.com/kubernetes/kubernetes/pull/117792/commits/9010304aec81062e34363fa69d653bdb03a86337